src/common/k8s-api/kube-api.ts

```diff
@@ -262,25 +289,20 @@ export class KubeApi<T extends KubeObject> {
   protected resourceVersions = new Map<string, string>();
   protected watchDisposer: () => void;
   private watchId = 1;
+  protected options: Pick<IKubeApiOptions<T>, "apiBase" | "checkPreferredVersion" | "fallbackApiBases">;
 
-  constructor(protected options: IKubeApiOptions<T>) {
-    const {
-      objectConstructor,
-      request = apiKube,
-      kind = options.objectConstructor?.kind,
-      isNamespaced = options.objectConstructor?.namespaced,
-    } = options || {};
-
+  constructor({ objectConstructor, request, kind, isNamespaced, ...options }: IKubeApiOptions<T>) {
```

Comment:
`options` is the right type just by virtue of the other fields being destructured here?

Reply:
Do you mean the right name?

Reply:
I mean if you didn't destructure `kind`, say, then `options` is not the expected type. Just seems odd to rely on `...`

Reply:
But I realized that all this is a breaking change, so I will revert this options change